### PR TITLE
Fix(Modules/1v1-Arena): Fix bugs with 1v1 queue system, Battlemaster NPC Crashing and Ready Mark Crystals

### PR DESCRIPTION
**Pull request by Rastrian**
## Changes Proposed:
-  Add structure to arena 1v1 queue and type
-  This fix the current script proposed on the module system, didn't affecting the blizzlike arena system (2v2, 3v3, 5v5 and skirmish modes).

## Issues Addressed:
- https://github.com/azerothcore/mod-1v1-arena/issues/27#issuecomment-862779969

## Tests Performed:

- Builded without errors with docker and manually
- Tested in-game with the arena 1v1 module
- Tested in-game the blizzlike arena models (2v2, 3v3 and 5v5)

## How to Test the Changes:

1. Make an fresh install of azerothcore and add arena1v1 module
2. Change the DEAFAULT Arena1v1.ArenaSlotID option to 4 in the module config (1v1arena.conf.dist), to avoid problems.
3. Test in-game

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Rastrian** [comment]
Some images of the fix working:

File: src/server/game/Battlegrounds/ArenaTeam.h [hunk 1]
```diff
@@ -120,7 +122,7 @@ struct ArenaTeamStats
     uint32 Rank;
 };
 
-#define MAX_ARENA_SLOT 3                                    // 0..2 slots
+#define MAX_ARENA_SLOT 4                                    // 0..2 slots
```
Comment (FrancescoBorzi) [inline src/server/game/Battlegrounds/ArenaTeam.h:125]: I guess the comment `0..2 slots` should be either removed or updated

**Comment by FrancescoBorzi** [comment]
good job @Rastrian ! I just left a minor comment

File: src/server/game/Battlegrounds/ArenaTeam.h (hunk 1 above)
Comment (Rastrian) [reply src/server/game/Battlegrounds/ArenaTeam.h:125]: Oh, i didn't see this, i will update.

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
Just use hooks for this

**Comment by Rastrian** [comment]
> Just use hooks for this, no need to add custom in the core, for custom changes there are modules

The current hooks doesnt allow ArenaType and BattlegroundQueueTypeId manipulation.

**Comment by Rastrian** [comment]
I found a solution, will be posting on module repository, thx guys.

**Pull request closed without merging**